python-sdk/src/astro/files/locations/azure/wasb.py

```diff
@@ -91,3 +93,21 @@ def openlineage_dataset_name(self) -> str:
         https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md
         """
         return urlparse(self.path).path
+
+    def get_new_path_for_snowflake_stage(self) -> str:
+        """
+        Get the altered path if needed for stage creation in snowflake stage creation. We need to modify the path since
+         Snowflake only accepts paths of format for stage creation:
+         "azure://<storage_account>.blob.core.windows.net/<container_name>/load/files/"
+         But SDK accepts paths
+         "wasb://<container_name>/<filename>" or "wasbs://<container_name>/<filename>"
+         To bridge the gap we use this method
+        """
+        if not check_required_option(self.load_options, "storage_account"):
+            raise ValueError(
+                f"Required param missing 'storage_account', pass {self.LOAD_OPTIONS_CLASS_NAME}"
+                f"(storage_account=<account_name>) to load_options"
+            )
+        url = urlparse(self.path)
+        azure_host = "blob.core.windows.net"
+        return f"{FileLocation.AZURE}://{self.load_options.storage_account}.{azure_host}/{url.netloc}/"  # type: ignore
```

Comment:
@tatiana We are already checking for `None` and `load_options` not having `storage_account` by function `check_required_option` and MyPy still raises a type error. Is there a better way to deal with this?

Reply:
Would it be worth to use `urllib.parse.urlunparse` instead of pure string concatenation?

Reply:
@utkarsharma2 did you try doing
```suggestion
        return f"{FileLocation.AZURE}://{str(self.load_options.storage_account)}.{azure_host}/{url.netloc}/"
```

Reply:
Using `str(self.load_options.storage_account)` will still raise the same exception right? Since the attribute may or not be present even for str() function